File: core/src/test/java/de/qabel/core/repository/inmemory/InMemoryTransactionManager.java
package de.qabel.core.repository.inmemory;

import de.qabel.core.repository.RunnableTransaction;
import de.qabel.core.repository.Transaction;
import de.qabel.core.repository.TransactionManager;
import de.qabel.core.repository.exception.PersistenceException;
import de.qabel.core.repository.exception.TransactionException;

import java.util.concurrent.Callable;

public class InMemoryTransactionManager implements TransactionManager {
    @Override
    public Transaction beginTransaction() throws TransactionException {
        return null;
    }

    @Override
    public <T> T transactional(Callable<T> transactionBasedCallback) throws PersistenceException {
        try {
            return transactionBasedCallback.call();
        } catch (Exception e) {
            throw new PersistenceException(e.getMessage(), e);
        }
    }

    @Override
    public void transactional(RunnableTransaction runnable) throws PersistenceException {
        try {
            runnable.run();
        } catch (Exception e) {
            throw new PersistenceException(e.getMessage(), e);
        }
    }
}

File: core/src/test/java/de/qabel/core/repository/inmemory/InMemoryTransactionManager.kt
package de.qabel.core.repository.inmemory

import de.qabel.core.repository.RunnableTransaction
import de.qabel.core.repository.Transaction
import de.qabel.core.repository.TransactionManager
import de.qabel.core.repository.exception.PersistenceException
import de.qabel.core.repository.exception.TransactionException

import java.util.concurrent.Callable

class InMemoryTransactionManager : TransactionManager {
    @Throws(TransactionException::class)
    override fun beginTransaction(): Transaction {
        throw TransactionException("no transaction")
    }

    @Throws(PersistenceException::class)
    override fun <T> transactional(transactionBasedCallback: Callable<T>): T {
        try {
            return transactionBasedCallback.call()
        } catch (e: Exception) {
            throw PersistenceException(e.message ?: "", e)
        }

    }

    @Throws(PersistenceException::class)
    override fun transactional(runnable: RunnableTransaction) {
        try {
            runnable.run()
        } catch (e: Exception) {
            throw PersistenceException(e.message ?: "", e)
        }

    }
}
